File: Codes/Bingran Li/Tree/104. Maximum Depth of Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int maxDepth(TreeNode root) {
        
        // solution 1: 
        if(root==null)
            return 0;
        if(root.left==null && root.right==null)
            return 1;
        else if(root.left==null)
            return maxDepth(root.right)+1;
        else if(root.right==null)
            return maxDepth(root.left)+1;
        else{
            int max;
            int right = maxDepth(root.right);
            int left = maxDepth(root.left);
             max = right>left ? right:left;
            // there used to be ------time exceed--->WHY????
            // max = maxDepth(root.right)>maxDepth(root.left)?maxDepth(root.right):maxDepth(root.left);
            return max+1;
        }
            
    }
}

File: Codes/Kan Zheng/C++/Tree/104. Maximum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root==NULL)
            return 0;
        else
        {
            int maxleft = maxDepth(root->left);
            int maxright = maxDepth(root->right);
            return (maxleft>maxright?maxleft:maxright)+1;
        }
    }
};